Motion refactoring

gnandretta opened the pull request:
Just for review, please don't pull it in :)

ohhgabriel committed:
rename motions state to state_name

ohhgabriel committed:
introduce state objects

ohhgabriel committed:
let a motion delegate assert_state to its state

ohhgabriel committed:
let a motion delegate permit? to its state

ohhgabriel committed:
we don't need Motion#publicly_visible? anymore

ohhgabriel committed:
add ScheduledMotionUpdate to handle resque jobs

ohhgabriel committed:
remove some duplication in the motion states

ohhgabriel committed:
remove the 'waitingexpedited' state

ohhgabriel committed:
remove 'objected' and 'waitingobjection' states

The 'discussing' state now replace those.  Also, now there's a
'objection' Event type, and that changes the concept of what
Motion#objected! was doing.

ohhgabriel committed:
remove 'passed' state, use just 'voting'

ohhgabriel committed:
change 'failed' and 'approved' states to 'closed'

ohhgabriel committed:
rename Motion#assert_state to update_state

ohhgabriel committed:
move resque enqueuing to the motion states

ohhgabriel committed:
oops, 'passed' is no longer a valid state name

ohhgabriel committed:
stop allowing nil as a state name

ohhgabriel committed:
clean some comments

We don't need them anymore since the code is already telling that :)

ohhgabriel committed:
be consistent when assigning motion attributes

ohhgabriel committed:
motion states refactor

ohhgabriel committed:
get rid of unnecessary temp

ohhgabriel committed:
these brackets shouldn't have been there, right?

thedelchop commented:
Motion should probably have a validator for member. Someone needs to own a motion.  We should also probably check to make sure that a motion's name is unique I would think, if not, things could get confusing.

ohhgabriel committed:
clean the database before running the specs

ohhgabriel committed:
move Member#can? specs to the motion states

ohhgabriel committed:
add some specs

ohhgabriel committed:
remove Motion#waitingexpedited!

ohhgabriel committed:
remove Motion#waitingobjection!

ohhgabriel committed:
remove Motion#objected!

ohhgabriel committed:
remove Motion#passed!

ohhgabriel committed:
remove Motion#approved!

ohhgabriel committed:
remove Motion#failed!

ohhgabriel committed:
remove Motion#waitingsecond!

ohhgabriel committed:
dry Motion up a little

ohhgabriel committed:
confidently remove Motion#update_state specs

We're already testing the same in the motion states, where the action is
happening.

ohhgabriel committed:
add 'Application updates motion states' feature

ohhgabriel committed:
set 'waitingsecond' as the default state name

ohhgabriel committed:
remove original state change specs

This is already tested in the motion state objects, and now we have a
cucumber spec for it too.

ohhgabriel committed:
use validates instead of validates_inclusion_of

Because validates_inclusion_of will be deprecated soon.

ohhgabriel committed:
Merge remote branch 'wycats/master' into motion-refactoring-merge

Conflicts:
	app/models/event.rb
	app/models/motion.rb
	db/schema.rb
	features/step_definitions/member_steps.rb
	spec/models/member_spec.rb
	spec/spec_helper.rb

gnandretta commented:
This is unmergeable, replaced by https://github.com/wycats/jquery-governance/pull/48
